src/Application.php

```diff
@@ -325,10 +325,17 @@ public function initProviders()
         // Make sure we keep our current locale..
         $currentlocale = $this['locale'];
 
-        // Setup Swiftmailer, with optional SMTP settings. If no settings are provided in config.yml, mail() is used.
+        // Setup Swiftmailer, with the selected Mail Transport options: smtp or `mail()`.
         $this->register(new Silex\Provider\SwiftmailerServiceProvider());
-        if ($this['config']->get('general/mailoptions')) {
+        if ($this['config']->get('general/mailoptions/transport') == 'mail') {
+            // Use the 'mail' transport.
+            $this['swiftmailer.transport'] = \Swift_MailTransport::newInstance();
```

Comment:
Reading this, I *think* it's nog bad to set this: http://silex.sensiolabs.org/doc/providers/swiftmailer.html#usage-in-commands